echopype/utils/uwa.py

```diff
@@ -56,73 +64,91 @@ def calc_sound_speed(
+    formula_source: str, {"AM", "FG", "AZFP"}
+        Source of formula used to calculate sound speed.
+        "AM" uses the formula from Ainslie and McColm (1998).
+        "FG" uses the formula from Francois and Garrison (1982).
+        "AZFP" uses the the formula supplied in the AZFP Matlab code.
+        See Notes below for the references.
 
     Returns
     -------
-    Sea absorption [dB/m]
+    Seaw ater absorption [dB/m].
+
+    Notes
+    -----
+    Ainslie MA, McColm JG. (1998). A simplified formula for viscous
+    and chemical absorption in sea water.
+    The Journal of the Acoustical Society of America, 103(3), 1671–1672.
+    https://doi.org/10.1121/1.421258
+
+    Francois RE, Garrison GR. (1982). Sound absorption based on
+    ocean measurements. Part II: Boric acid contribution and equation
+    for total absorption.
+    The Journal of the Acoustical Society of America, 72(6), 1879–1890.
+    https://doi.org/10.1121/1.388673
+
+    The accuracy of the simplified formula from Ainslie & McColm 1998
+    compared with the original complicated formula from Francois & Garrison 1982
+    was demostrated between 100 Hz and 1 MHz.
     """
-    # fmt: off
     if formula_source == "FG":
         f = frequency / 1000.0
-        d = distance / 1000.0
+        d = 1 / 1000.0  # ref to 1 m
         c = 1412.0 + 3.21 * temperature + 1.19 * salinity + 0.0167 * pressure
         A1 = 8.86 / c * 10 ** (0.78 * pH - 5)
         P1 = 1.0
         f1 = 2.8 * np.sqrt(salinity / 35) * 10 ** (4 - 1245 / (temperature + 273))
         A2 = 21.44 * salinity / c * (1 + 0.025 * temperature)
         P2 = 1.0 - 1.37e-4 * pressure + 6.2e-9 * pressure * pressure
-        f2 = 8.17 * 10 ** (8 - 1990 / (temperature + 273)) / (1 + 0.0018 * (salinity - 35))  # noqa
+        f2 = (
+            8.17
+            * 10 ** (8 - 1990 / (temperature + 273))
+            / (1 + 0.0018 * (salinity - 35))
+        )
         P3 = 1.0 - 3.83e-5 * pressure + 4.9e-10 * pressure * pressure
         if temperature < 20:
             A3 = (
                 4.937e-4
                 - 2.59e-5 * temperature
-                + 9.11e-7 * temperature ** 2
-                - 1.5e-8 * temperature ** 3
+                + 9.11e-7 * temperature * temperature
+                - 1.5e-8 * temperature * temperature * temperature
             )
         else:
             A3 = (
                 3.964e-4
                 - 1.146e-5 * temperature
-                + 1.45e-7 * temperature ** 2
-                - 6.5e-10 * temperature ** 3
+                + 1.45e-7 * temperature * temperature
+                - 6.5e-10 * temperature * temperature * temperature
             )
         a = (
             A1 * P1 * f1 * f * f / (f1 * f1 + f * f)
             + A2 * P2 * f2 * f * f / (f2 * f2 + f * f)
             + A3 * P3 * f * f
         )
-        # convert to db/m from db/km
-        sea_abs = -20 * np.log10(10 ** (-a * d / 20.0)) / 1000
+        sea_abs = a * d
```

Comment:
Wow, that's way simpler! I had to convert the old formula by hand to confirm that they're the same -- except for the `/1000` that you've dropped, but it looks like it all balances out by the change to the `d` formula

Reply:
I didn't actually dive into the code (not from me!) but took the lazy route to fix bugs via testing... Looking at this again, I'd rather things as close to the original paper as possible instead of this fix at the end. The original output (in the paper) was in units [dB / km] so it the scaling should just be at the end. I've fix this one and will simplify the formula too: it is pretty awful to have things like `temperature * temperature * temperature`.

Reply:
Revised!